cmake/testing.cmake

```diff
@@ -2,30 +2,30 @@ include(CMakeParseArguments)
 enable_testing()
 set(RUNNER "${PROJECT_SOURCE_DIR}/tools/cmake_test.py")
 option(RP_ENABLE_TESTS "Useful for disabling all tests" ON)
-option(RP_ENABLE_INTEGRATION_TESTS "control if integrations are bulit and ran" ON)
+option(RP_ENABLE_FIXTURE_TESTS "control if integrations are bulit and ran" ON)
```

Comment:
i'm not too sure i understand this name change. it seems to be trying to equate "fixture" tests with a specific type of fixture, one that spins up the whole application stack which presumably we want to handle differently. but other tests that use fixtures in their implementation don't use the whole app fixture.

Reply:
Yeah, this is one of those "naming is hard" situations, but if we can think of something snappy to replace 'fixture' here I'm up for it.

I was mainly just match our existing slang around these tests, but of course it's true that there are plenty of fixtures which aren't "the big redpanda fixture".

Reply:
I guess the other question is why _not_ call them "integration" tests... my motivation there is to preserve "integration testing" to describe testing a feature in the usual environment where the components get integrated together (i.e. built into a `redpanda`).

Some folks use integration testing to describe any test that involves >1 component, but I find that a bit too generous (if someone has a C++ test that spins up two classes, I'm not going sign off on "this feature is integration tested").

Reply:
I agree with all this, and i'm good with fixture or integration until we choose to revisit some of the testing infra.

--

IMO we are being forced to a choose a name here because the test wrapper forces that choice, and in reality we just have two: the boost unit test framework and the seastar benchmark framework. Everything else is a categorization that can be selected on / queried through labels etc...